src/setup/dependencies.ts

```diff
@@ -71,4 +80,10 @@ export interface Dependencies {
     removeItem: (key: string) => Promise<void>
     clear: () => Promise<void>
   }
+  auth: {
+    init: (options: InitOptions) => Promise<State | null>
+    register: (options: { email: string; username: string }) => Promise<{ success: boolean }>
+    isUsernameValid: (username: string) => Promise<boolean>
+    isUsernameAvailable: (username: string) => Promise<boolean>
```

Comment:
Just a Q for clarification: Why do we abstract these functions? We're using the same implementation for both local auth as well as auth-lobby based auth it seems.
Is this so that someone would in principle be able to implement their own "username database"? If so, is that possible today? I think we'd need email-like identifiers for that, i.e. something like `matheus23@fission.codes` usernames instead of `matheus23`. E.g. in the DNSLink fetching code.
Maybe it doesn't hurt having these abstractions in here today already, but I'm not sure. :thinking:

Reply:
Yes, this is so that you can plug in any arbitrary lookup method, rather than forcing everyone to make it look like Fission's. For example, web3.storage has this in a big database table that's at a different endpoint.

Reply:
RE validation: services may have special validation rules for their service (e.g. no curse words in Spanish). I believe that the plan is to provide defaults and let users override them if needed.

> something like matheus23@fission.codes usernames instead of matheus23

That's more of an interoperability question. Blaine calls this style the "Jesus of Nazareth" format, and I agree that it reads very well (in English at least)

Reply:
(But per service, you can imply the host, in the same way that Discord hides the nonce in your username unless there's a collision on that server)